src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -671,6 +682,27 @@ bool FileBasedWal::rollbackToLog(LogID id) {
     return true;
 }
 
+void FileBasedWal::cleanWAL() {
+    std::lock_guard<std::mutex> g(walFilesMutex_);
+    if (walFiles_.empty()) {
+        return;
+    }
+    auto now = time::WallClock::fastNowInSec();
+    // We skip the latest wal file because it is beging written now.
+    size_t index = 0;
+    auto it = walFiles_.begin();
+    auto size = walFiles_.size();
+    while (it != walFiles_.end()) {
+        if (index++ < size - 1 &&  (now - it->second->mtime() > policy_.ttl)) {
+            VLOG(1) << "Clean wals, Remove " << it->second->path();
+            unlink(it->second->path());
+            it = walFiles_.erase(it);
+        } else {
+            it++;
```

Comment:
maybe ++iter is better

Reply:
It would be better to use "remove_if+erase".

Reply:
Good point